WordPress/gutenberg
# Add hovered block store and show hovered template part's label in edit site header component.

## Description


Adds a way to track hovered blocks by adding them as an array of clientIds in the `block-editor` store.  This information is then used to show the hovered template part's label in the component added in #25320 (alternative approaches discussed in #25469).

![hovered-template-part-label](https://user-images.githubusercontent.com/28742426/93951844-7195ba80-fd15-11ea-852f-46347ab10341.gif)

The first approach to implement this store tried to add/remove blocks individually by clientId.  However, this ran into a blocker with the inserter triggering removal for all blocks that it appeared to be within the bounds of.  Parsing the blocks found under the cursor and setting the full list solves this problem, and the hovered name is no longer disrupted by the inserter as shown in the gif above. 

## How has this been tested?



* Tested in the Site Editor
   * hover over a template part and verify its label is displayed in the middle of the top toolbar.
   * hover over a block inserter within the template part and verify the label in the top toolbar does not disappear.
* Smoke tested in Post Editor.

## Screenshots 

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
